packages/webpack/src/buildtime/generator.js

```diff
@@ -51,130 +51,154 @@ function processRequirements(requirements, module) {
+        diag.rawDebug(5, {
+          wrappingEnabed,
+          module,
+          options,
+        })
 
-      // originalGenerate adds requirements to options.runtimeRequirements. runtimeKit needs to be derived from those.
-      // We also depend on __webpack_require__ being there, so let's add it
-      options.runtimeRequirements.add('__webpack_require__')
-
-      // Turn off "use strict" being added in front of modules on final wrapping by webpack.
-      // If anything attempts to reverse it, we want to ignore it
-      if (
-        module.buildInfo && // never found an actual case in which buildInfo wasn't there, but types suggest it's possible
-        (module.buildInfo.strict === true ||
-          module.buildInfo.strict === undefined) // seems like it's possible for generation to run more than once for the same module
-      ) {
-        Object.defineProperty(module.buildInfo, 'strict', {
-          get: () => false,
-          set: () => {
-            // TODO: make the error more informative - explaining why the attempt to strict mode had to be skipped here but is applied anyway
-            console.warn(
-              'Attempted to set strict mode on module',
-              module.rawRequest,
-              Error().stack
-            )
-          },
+        // used to be .apply(this, arguments) but typescript complained. I feel it's worse that way
+        const originalGeneratedSource = originalGenerate.call(
+          this,
+          module,
+          options
+        )
+
+        // skip doing anything if marked as excluded by the excludeLoader
+        // this may clean up the `tooEarly` list too
+        if (isExcluded(module)) {
+          excludes.push(module.userRequest)
+          diag.rawDebug(3, `skipped wrapping ${module.rawRequest}`)
+          return originalGeneratedSource
+        }
+
+        if (!wrappingEnabed) {
+          tooEarly.push(module.request || module.rawRequest)
+          return originalGeneratedSource
+        } else if (!PROGRESS.done('pathsProcessed')) {
+          throw Error(
+            'LavaMoatPlugin: attempting to wrap generated module before all paths from bundle were known.'
+          )
+        }
+
+        // originalGenerate adds requirements to options.runtimeRequirements. runtimeKit needs to be derived from those.
+        // We also depend on __webpack_require__ being there, so let's add it
+        options.runtimeRequirements.add('__webpack_require__')
+
+        // Turn off "use strict" being added in front of modules on final wrapping by webpack.
+        // If anything attempts to reverse it, we want to ignore it
+        if (
+          module.buildInfo && // never found an actual case in which buildInfo wasn't there, but types suggest it's possible
+          (module.buildInfo.strict === true ||
+            module.buildInfo.strict === undefined) // seems like it's possible for generation to run more than once for the same module
```

Comment:
```suggestion
          (module.buildInfo.strict !== false) // seems like it's possible for generation to run more than once for the same module
```

I know this is not quite equivalent but may be feasible

Reply:
It's valid, but not what I was aiming to say to the developer looking at it in the future. 
Regardless of whether it's true or unset, we want to step in. 

Now I'm thinking maybe we always want to step in...